src/rgw/rgw_kafka.cc

```diff
@@ -166,7 +171,7 @@ void message_callback(rd_kafka_t* rk, const rd_kafka_message_t* rkmessage, void*
   if (tag_it != callbacks_end) {
       ldout(conn->cct, 20) << "Kafka run: n/ack received, invoking callback with tag=" << 
           *tag << " and result=" << rd_kafka_err2str(result) << dendl;
-      tag_it->cb(result);
+      tag_it->cb(-rd_kafka_err2errno(result));
```

Comment:
@yuvalif we did testing with these changes, and i feel we should not map the kafka error's to linux errors.
i do not see a reason on why we want to to convert these error's to linux error's, coz currently for any reason if there is any config issue with kafka, you get to know the error code on what the error is in the callback return code.
with error transition to linux based error, the only way to know the error is by turning on the debug level 20 which then prints the kafka error string `rd_kafka_err2str`.
i do not see any value by converting them linux error's.

Reply:
the reason for conversing to linux errors is that the calling code is not kafka specific.
e.g. we can reply them back to the client ion case of non-persistent notifications.

agree that we lose valuable info here (especially in case of persistent notification). 
will fix that, so we log the actual kafka errors

Reply:
making sure that the kafka errors are printed at log level 1